### PR TITLE
fix(timeseries-data-export): csv export limit

### PR DESCRIPTION
**Pull request by mrtraser**
PR based on #641 published by @annitya. The purpose of opening it – to move PR to gerabox.js repo and speed up refactoring.

### CODE REVIEW

**Comment by mrtraser** [comment]
@annitya, I'm currently working on it, cause rn result of exporting 2 and more timeseries produce the wrong data format in CSV file

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-647.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-647.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-647.gearboxjs.preview.cogniteapp.com

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 1]
```diff
@@ -36,31 +31,15 @@ import {
 type TimeseriesDataExportFormProps = TimeseriesDataExportProps &
   FormComponentProps;
 
-// TODO: Check tree shacking for TimeseriesDataExport component
+// TODO: Check tree shaking for TimeseriesDataExport component
 const { RangePicker } = DatePicker;
 const CELL_LIMIT = 10000;
-const formatData = 'YYYY-MM-DD HH:mm:ss';
+const formatData = 'YYYY-MM-DD_HH:mm:ss';
```
Comment (annitya) [inline src/components/TimeseriesDataExport/TimeseriesDataExport.tsx:37]: I'm guessing this change was on purpose, but could you perhaps satisfy my curiosity as to why it was made? :)

**Review by annitya — APPROVED** [review]

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 2]
```diff
@@ -118,11 +93,92 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
...
+      Number(end)
+    );
+
+    const startResults: (
+      | DatapointsGetAggregateDatapoint
+      | DatapointsGetDatapoint)[] = await context!.datapoints.retrieve({
+      ...request,
+      start,
+      end,
+      limit: 1,
```
Comment (polomani) [inline src/components/TimeseriesDataExport/TimeseriesDataExport.tsx:124]: this is a bit weirdly formatted

**Review by polomani — COMMENTED** [review]
lg, some minor things

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 3]
```diff
@@ -118,11 +93,92 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
...
+  ): Promise<TimeRange<number>> => {
+    let { start = 0, end = 0 } = request;
+    const items = request.items.map(item => ({
+      id: (item as DatapointsQueryId).id,
+    }));
+    const endResults = await context!.datapoints.retrieveLatest(items);
+
+    end = Math.min(
+      Math.max(
+        ...endResults.map(({ datapoints: [item] }) => item.timestamp.getTime())
```
[the remark this answers is not in the document]
Comment (polomani) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx:113]: there is one more below

[the remark this answers is not in the document]
Comment (polomani) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]: fixed

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 4]
```diff
@@ -118,11 +93,92 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
...
+      ),
+      Number(end)
+    );
+
+    const startResults: (
+      | DatapointsGetAggregateDatapoint
+      | DatapointsGetDatapoint)[] = await context!.datapoints.retrieve({
```
[the remark this answers is not in the document]
Comment (polomani) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]: fixed

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx [hunk 5]
```diff
@@ -118,11 +93,92 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
...
+    const requests = chunks
+      .map(([chunkStart, chunkEnd]) => ({
+        ...request,
+        start: chunkStart,
+        end: chunkEnd,
+        limit,
+      }))
+      .map(params => context!.datapoints.retrieve(params));
+
+    const datapoints: DatapointsGetAggregateDatapoint[][] = await Promise.all(
```
[the remark this answers is not in the document]
Comment (polomani) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]: fixed

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx (hunk 2 above)
Comment (polomani) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx]: fixed

File: src/components/TimeseriesDataExport/TimeseriesDataExport.tsx (hunk 1 above)
Comment (mrtraser) [reply src/components/TimeseriesDataExport/TimeseriesDataExport.tsx:36]: space breaks CSV formatting for different viewers, MS Excel for instance

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-647.gearboxjs.preview.cogniteapp.com

**Review by polomani — APPROVED** [review]

**Comment by polomani** [comment]
@annitya just to shed some lights on. We had to close the previous pr and reopen it here on gearbox repo instead. We did that just to be able to polish it and release faster.
The code is based on yours, so it is still your contribution, just a bit refactored.
Hope that's ok

**Comment by annitya** [comment]
I'm thankful for the extra help and speedy release. Very much appreciated :-)

**Pull request merged**